### PR TITLE
Ev 231 add pan sharpen

### PR DESCRIPTION
**Pull request by tculp**

### CODE REVIEW

File: plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy [hunk 1]
```diff
@@ -353,21 +351,20 @@ class WebMappingService implements InitializingBean
          location: bboxMidpoint
       ]
 
-      if(addLocation)
-      {
-        logParams.location = bboxMidpoint
```
Comment (arocnies) [inline plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: We use this for our metrics and heatmap. Did this get removed entirely?

Comment (Spectrality) [reply plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: looks to me like just the addLocation boolean got removed, which seems reasonable. Why would you not want it logged always? the bboxMidpoint seems to still be calculated. Although I have a question about that, which I'll mention in the code

File: plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy [hunk 2]
```diff
@@ -314,7 +318,6 @@ class WebMappingService implements InitializingBean
         double scaleY = wmsParams.height.toDouble()/(bbox.maxY-bbox.minY)
         bboxMidpoint = [y: (bbox.minY + bbox.maxY) / 2, x: (bbox.minX + bbox.maxX) / 2]
```
Comment (Spectrality) [inline plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: why does bboxMidpoint here look like {y: blah, x: blah} but below it is {lat: blah, lon: blah} ? won't that mess up the parsing of the logs as far as location goes, @arocnies ?

Comment (Spectrality) [reply plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: also, @tculp , I recognize this is not code you added. just wondering if it is something we should change

File: plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy [hunk 3]
```diff
@@ -275,32 +276,35 @@ class WebMappingService implements InitializingBean
...
     def bboxMidpoint
     def result
-    Boolean addLocation = true
 
     Map<String, Object> omsParams = parseLayers( wmsParams )
 
     if ( omsParams )
     {
+      if (getPsm && ! omsParams.hasProperty('images[1].file')) {
+          throw new Exception('The filter given to the getPsm call did not contain at least two images')
```
Comment (Spectrality) [inline plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: how does this exception get handled?

Comment (arocnies) [reply plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: I suspect Grails catches this and throws back an error in the response?

File: plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy (hunk 1 above)
Comment (tculp) [reply plugins/omar-wms-plugin/grails-app/services/omar/wms/WebMappingService.groovy]: Nope, yo code was duped

File: plugins/omar-wms-plugin/grails-app/controllers/omar/wms/WmsController.groovy [hunk 4]
```diff
@@ -198,6 +201,128 @@ where:
...
+		def outputStream = null
+		try
+		{
+			outputStream = response.outputStream
+			if(wmsParams.validate())
+			{
+				def result = webMappingService.getMap( wmsParams, true )
+				if(result.status) response.status = result.status
+				if(result.contentType) response.contentType = result.contentType
+				if(result.buffer?.length) response.contentLength = result.buffer.length
```
[the remark this answers is not in the document]
Comment (tculp) [reply plugins/omar-wms-plugin/grails-app/controllers/omar/wms/WmsController.groovy]: Yep

**Pull request merged**